### PR TITLE
Add SQL Server-specific isolation levels

### PR DESCRIPTION
**Pull request by uaihebert**
add a transaction isolation level, specific for mssql server.

#19

### CODE REVIEW

File: src/main/java/io/r2dbc/mssql/MssqlIsolationLevel.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    REPEATABLE_READ("REPEATABLE READ"),
+
+    /**
+     * The serializable isolation level.
+     */
+    SERIALIZABLE("SERIALIZABLE"),
+
+    /**
+     * The serializable isolation level.
```
Comment (mp911de) [inline src/main/java/io/r2dbc/mssql/MssqlIsolationLevel.java]: Nit: The comment should be: `snapshot isolation level`.

File: src/main/java/io/r2dbc/mssql/MssqlConnection.java [hunk 2]
```diff
@@ -197,7 +204,7 @@ Client getClient() {
         return this.client;
     }
 
-    private static String getIsolationLevelSql(IsolationLevel isolationLevel) {
+    private static String getIsolationLevelSql(MssqlIsolationLevel isolationLevel) {
```
Comment (mp911de) [inline src/main/java/io/r2dbc/mssql/MssqlConnection.java]: We could probably get rid of this method and use `MssqlIsolationLevel.asSql()`. What do you think?

File: src/main/java/io/r2dbc/mssql/MssqlIsolationLevel.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.mssql;
+
+import io.r2dbc.mssql.util.Assert;
+
+public enum MssqlIsolationLevel {
```
Comment (mp911de) [inline src/main/java/io/r2dbc/mssql/MssqlIsolationLevel.java]: We should add some Javadoc here mentioning that this class contains specific extensions to SQL Server and add a reference (`@see …`) to R2DBC's `IsolationLevel` enum.

File: src/main/java/io/r2dbc/mssql/MssqlIsolationLevel.java (hunk 1 above)
Comment (uaihebert) [reply src/main/java/io/r2dbc/mssql/MssqlIsolationLevel.java]: ops.

File: src/main/java/io/r2dbc/mssql/MssqlConnection.java (hunk 2 above)
Comment (uaihebert) [reply src/main/java/io/r2dbc/mssql/MssqlConnection.java]: totally agree.

**Comment by uaihebert** [comment]
Hello @mp911de 

all done

**Comment by mp911de** [comment]
Thanks a lot. Care to squash your changes to a single commit?

**Comment by uaihebert** [comment]
hello @mp911de 

I just squashed all the commits. thank you for the support

**Comment by uaihebert** [comment]
do I need to do something else? or will this be merged on the next release?

**Comment by mp911de** [comment]
Thanks a lot. I'll take the pull request from here. The changes will be released with 1.0 M8, probably end of March.

**Pull request merged**